src/v/pandaproxy/server.cc

```diff
@@ -83,7 +83,7 @@ struct handler_adaptor : ss::httpd::handler_base {
       std::unique_ptr<ss::request> req,
       std::unique_ptr<ss::reply> rep) final {
         auto measure = _probe.auto_measure();
-        auto guard = gate_guard(_pending_requests);
```

Comment:
is gate_guard something we can remove from the tree now? not that we need to do that in this PR. just curious.